lib/ansible/modules/cloud/vmware/vmware_vmkernel_facts.py

```diff
@@ -0,0 +1,210 @@
+                vmks_list = self.query_service_type_for_vmks(host, service_type)
+                self.service_type_vmks[host.name][service_type] = vmks_list
+
+    def query_service_type_for_vmks(self, host_system, service_type):
+        """
+        Function to return list of VMKernels
+        Args:
+            host_system: Host system managed object
+            service_type: Name of service type
+
+        Returns: List of VMKernel which belongs to that service type
+
+        """
+        vmks_list = []
+        query = None
+        try:
+            query = host_system.configManager.virtualNicManager.QueryNetConfig(service_type)
+        except vim.fault.HostConfigFault as config_fault:
+            self.module.fail_json(msg="Failed to get all VMKs for service type %s due to"
+                                      " host config fault : %s" % (service_type, to_native(config_fault.msg)))
+        except vmodl.fault.InvalidArgument as invalid_argument:
+            self.module.fail_json(msg="Failed to get all VMKs for service type %s due to"
+                                      " invalid arguments : %s" % (service_type, to_native(invalid_argument.msg)))
+        except Exception as e:
+            self.module.fail_json(msg="Failed to get all VMKs for service type %s due to"
+                                      "%s" % (service_type, to_native(e)))
+
+        if not query.selectedVnic:
+            return vmks_list
+        selected_vnics = [vnic for vnic in query.selectedVnic]
+        vnics_with_service_type = [vnic.device for vnic in query.candidateVnic if vnic.key in selected_vnics]
+        return vnics_with_service_type
+
+    def gather_host_vmk_facts(self):
+        hosts_facts = {}
+
+        for host in self.hosts:
+            host_vmk_facts = []
+            host_network_system = host.config.network
+            if host_network_system:
+                vmks_config = host.config.network.vnic
+                for vmk in vmks_config:
+                    host_vmk_facts.append(dict(
+                        device=vmk.device,
+                        key=vmk.key,
+                        portgroup=vmk.portgroup,
+                        ipv4_address=vmk.spec.ip.ipAddress,
+                        ipv4_subnet_mask=vmk.spec.ip.subnetMask,
+                        dhcp=vmk.spec.ip.dhcp,
+                        mac=vmk.spec.mac,
+                        mtu=vmk.spec.mtu,
+                        stack=vmk.spec.netStackInstanceKey,
+                        enable_vsan=True if vmk.device in self.service_type_vmks[host.name]['vsan'] else False,
+                        enable_vmotion=True if vmk.device in self.service_type_vmks[host.name]['vmotion'] else False,
+                        enable_management=True if vmk.device in self.service_type_vmks[host.name]['management'] else False,
+                        enable_ft=True if vmk.device in self.service_type_vmks[host.name]['faultToleranceLogging'] else False,
```

Comment:
`True if foo in bar else False` → `(foo in bar)`
it's already a boolean

Reply:
ping @Akasurde

Reply:
OK.

Reply:
When you think too much about code, these kinda of thing will happen. :)